config/katello.migrations/170331152302-add-cli.rb

```diff
@@ -1 +1,3 @@
-answers['foreman::cli'] ||= true
+unless answers.key?('foreman::cli')
+  answers['foreman::cli'] = true
+end
```

Comment:
I am guessing there is some subtle difference here. Can you elaborate on the change for this older migration?

Reply:
What `||=` does is that if the value evaluates `false` (so `false` or `nil`), it sets the value to `true`.

So previously this happened:
```ruby
answers['foreman::cli'] = false # from the answers
answers['foreman::cli'] ||= true # from the migration
```

So the result was that it was always enabled while the intended migration was to add it with a default value of `true`.

In https://github.com/theforeman/foreman-installer/pull/700 I played around with implementing a hypothetical DSL for migrations that would convey the intention of the migration much better.

Reply:
Ah, I understood the behavior of `||=` with `nil` but not `false`. Now I see why this was broken. TY